ando/tools/generator/bidsconverter.py

```diff
@@ -1,33 +1,231 @@
+                    file=self._file_name_dict)
+
+    def _write_participant_tsv(self):
+        self._write_tsv(
+            self._create_participant_df()['data'],
+            self.output_path/self._create_participant_df()['name'])
+
+
+    @staticmethod
+    def _write_tsv(data, write_path):
+        assert isinstance(data, pd.Dataframe), f'{data} should be a df'
+        if not write_path.exists():
+            data.dropna(axis='columns', how='all', inplace=True)
+            data.to_csv(write_path, sep='\t', index=False)
+
+    @staticmethod
+    def _write_json(data, write_path):
+        assert isinstance(data, dict), f'{data} should be a dict'
+        if not write_path.exists():
+            with open(write_path) as j:
+                json.dump(data,j)
+
+    @staticmethod
+    def _move_data_file(source, dest, move=True):
+        if move:
+            if not source.exists():
+                source.replace(dest)
+        else:
+            if not source.exists():
+                dest.symlink_to(source)
+
+    def organize(self, output_path=None, move_file=False,
+                 re_write=True, validate=True):
+        if output_path is not None:
+            self.output_path = Path(output_path)
+        if re_write and self.output_path.exists():
+            shutil.rmtree(self.output_path)
+        self.output_path.mkdir(parents=True, exist_ok=True)
+        data, loc = self._parse_data_dict(
+            self._participants_dict,
+            self.output_path)
+        self._write_tsv(data, loc)
+        data, loc = self._parse_data_dict(
+            self._dataset_desc_json,
+            self.output_path)
+        self._write_json(data, loc)
+        for participant, sessions in self._labels_dict.items():
+            data, loc = self._parse_data_dict(
+                self._sessions_dict[participant],
+                self.output_path)
+            self._write_tsv(data, loc)
+            for session in sessions:
+                data, loc = self._parse_data_dict(
+                    self._channels_dict[participant][session],
+                    self.output_path)
+                self._write_tsv(data,loc)
+                data, loc = self._parse_data_dict(
+                    self._probes_dict[participant][session],
+                    self.output_path)
+                self._write_tsv(data, loc)
+                data, loc = self._parse_data_dict(
+                    self._contacts_dict[participant][session],
+                    self.output_path)
+                self._write_tsv(data, loc)
+                data, loc = self._parse_data_dict(
+                    self._ephys_dict[participant][session],
+                    self.output_path)
+                self._write_json(data, loc)
+                data, loc = self._parse_data_dict(
+                    self._file_name_dict[participant][session],
+                    self.output_path)
+                self._move_data_file(data, loc, move=move_file)
```

Comment:
Can you use a for loop here?

Reply:
I could loop through the different dictionaries (`_probes_dict`, `_contacts_dict` etc) but the last two use a different method also : `write_json` and `move_data_file`.